pkg/config/security/api_client.go

```diff
@@ -0,0 +1,335 @@
+			vc.Logger.Errorf("unable to get the API client; err=%s", err.Error())
+			return nil, "", err
+		}
+
+		vc.Logger.Errorf("unable to get the API client; code=%d, body=%s", response.StatusCode(), string(response.Body))
+		return nil, "", errorsx.G11NError("unable to get the API client with clientName %s; status=%d", clientName, response.StatusCode())
+	}
+
+	Client := &APIClientConfig{}
+	if err = json.Unmarshal(response.Body, Client); err != nil {
+		return nil, "", errorsx.G11NError("unable to get the API client")
+	}
+
+	return Client, response.HTTPResponse.Request.URL.String(), nil
+}
+
+func (c *ApiClient) GetAPIClients(ctx context.Context, search string, sort string, page int, limit int) (*APIClientListResponse, string, error) {
+	vc := contextx.GetVerifyContext(ctx)
+	client := openapi.NewClientWithOptions(ctx, vc.Tenant, c.Client)
+	params := &openapi.GetAPIClientsParams{}
+	if len(search) > 0 {
+		params.Search = &search
+	}
+	if len(sort) > 0 {
+		params.Sort = &sort
+	}
+
+	pagination := url.Values{}
+	if page > 0 {
+		pagination.Set("page", fmt.Sprintf("%d", page))
+	}
+
+	if limit > 0 {
+		pagination.Set("limit", fmt.Sprintf("%d", limit))
+	}
+
+	if len(pagination) > 0 {
+		paginationStr := pagination.Encode()
+		params.Pagination = &paginationStr
+	}
+
+	response, err := client.GetAPIClientsWithResponse(ctx, params, func(ctx context.Context, req *http.Request) error {
+		req.Header.Set("Accept", "application/json")
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", vc.Token))
+		return nil
+	})
+
+	if err != nil {
+		vc.Logger.Errorf("unable to get the API clients; err=%s", err.Error())
+		return nil, "", err
+	}
+
+	if response.StatusCode() != http.StatusOK {
+		if err := errorsx.HandleCommonErrors(ctx, response.HTTPResponse, "unable to get API clients"); err != nil {
+			vc.Logger.Errorf("unable to get the API clients; err=%s", err.Error())
+			return nil, "", err
+		}
+
+		vc.Logger.Errorf("unable to get the API clients; code=%d, body=%s", response.StatusCode(), string(response.Body))
+		return nil, "", errorsx.G11NError("unable to get the API clients")
+	}
+
+	apiclientsResponse := &APIClientListResponse{}
+	if err = json.Unmarshal(response.Body, &apiclientsResponse); err != nil {
+		vc.Logger.Errorf("unable to get the API clients; err=%s, body=%s", err, string(response.Body))
+		return nil, "", errorsx.G11NError("unable to get the API clients")
+	}
+
+	return apiclientsResponse, response.HTTPResponse.Request.URL.String(), nil
+}
+
+func (c *ApiClient) UpdateAPIClient(ctx context.Context, apiClientConfig *APIClientConfig) error {
+	vc := contextx.GetVerifyContext(ctx)
+	client := openapi.NewClientWithOptions(ctx, vc.Tenant, c.Client)
+	if apiClientConfig == nil {
+		vc.Logger.Errorf("client object is nil")
+		return errorsx.G11NError("client object is nil")
+	}
+
+	id, err := c.GetAPIClientId(ctx, apiClientConfig.ClientName)
```

Comment:
Why would the ID not be in the resource object? It should be a requirement for updates.